vortex-array/src/macros.rs

```diff
@@ -58,6 +58,10 @@ macro_rules! impl_encoding {
                             stats
                     )?)
                 }
+
+                pub fn maybe_from(data: $crate::ArrayData) -> Option<Self> {
```

Comment:
This is the main change

Reply:
Can you add a doc-string to the effect of "this is preferred over TryFrom when a backtrace isn't needed"